### PR TITLE
[WIP] Added support for filtering events by type

### PR DESCRIPTION
**Pull request by moysesb**
Missing integration tests and documentation.

Signed-off-by: Moysés Borges <moyses@gmail.com>

### CODE REVIEW

File: pylxd/client.py [hunk 1]
```diff
@@ -374,6 +385,14 @@ def events(self, websocket_client=None):
 
         client = websocket_client(self.websocket_url)
         parsed = parse.urlparse(self.api.events._api_endpoint)
-        client.resource = parsed.path
+
```
Comment (ajkavanagh) [inline pylxd/client.py]: As the function signature is being modified, the docstring ought to be updated to current best practice and document the types of the params and return types.  Could I suggest:

```Python
    """
    ...
    :param websocket_client: Optional parameter can be specified for implementation-specific handling of events as they occur.
    :type websocket_client: ws4py.client import WebSocketBaseClient
    :param event_types: dictionary of event types to propagate
    :type event_types: Dict[EventType]
    :returns: instance of the websocket client
    :rtype: Option[_WebsocketClient(), :param:`websocket_client`]
    """

File: pylxd/client.py [hunk 2]
```diff
@@ -354,7 +365,7 @@ def websocket_url(self):
         url = parse.urlunparse((scheme, host, '', '', '', ''))
         return url
 
-    def events(self, websocket_client=None):
+    def events(self, websocket_client=None, event_types=DEFAULT_EVENT_FILTER):
```
[the remark this answers is not in the document]
Comment (ajkavanagh) [reply pylxd/client.py]: Sorry, yes, I missed that it was a `Set` constructor rather than `dict`.  Making it a `frozenset` would be fine too.  Thanks.

**Comment by ajkavanagh** [comment]
There's just a minor whitespace failure on it;  if you are happy with the code, then please could you take the [WIP] prefix off, and then lets see if we can get it merged.  Thanks!

**Comment by moysesb** [comment]
Fixed the whitespace problem, but I still want to add an integration test. I'll probably get to it later today.

**Comment by ajkavanagh** [comment]
> Fixed the whitespace problem, but I still want to add an integration test. I'll probably get to it later today.

Okay, thanks, that would be great.

**Pull request merged**